crowbar_framework/app/controllers/backup_controller.rb

```diff
@@ -0,0 +1,172 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class BackupController < ApplicationController
+  skip_before_filter :enforce_installer
+
+  #
+  # Backup
+  #
+  # Provides the restful api call for
+  # /utils/backup 	GET 	Returns a json list of available backups
+  def index
+    @backups = Backup.all
+
+    respond_to do |format|
+      format.html { @backups }
+      format.json { render json: @backups }
+    end
+  end
+
+  #
+  # Backup
+  #
+  # Provides the restful api call for
+  # /utils/backup   POST   Trigger a backup
+  def backup
+    @backup = Backup.new(
+      name: params[:filename]
+    )
+    respond_to do |format|
+      if !@backup.nil? && @backup.save
+        format.json { head :ok }
+        format.html { redirect_to backup_path }
+      else
+        msg = I18n.t(".invalid_filename", scope: "backup.index")
+        format.json { render json: msg }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+      end
+    end
+  end
+
+  #
+  # Restore
+  #
+  # Provides the restful api call for
+  # /utils/backup/restore   POST   Trigger a restore
+  def restore
+    @backup = Backup.new(
+      name: params[:name],
+      created_at: params[:created_at]
+    )
+
+    respond_to do |format|
+      if @backup.valid?
+        ret = @backup.restore
+        if ret[:status] == :ok
+          format.json { head :ok }
+          format.html { redirect_to backup_path }
+        else
+          format.html do
+            flash[:alert] = ret[:msg]
+            redirect_to backup_path
+          end
+          format.json { render json: ret[:msg] }
+        end
+      else
+        msg = I18n.t(".invalid_backup", scope: "backup.index")
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+        format.json { render json: msg }
+      end
+    end
+  end
+
+  #
+  # Download
+  #
+  # Provides the restful api call for
+  # /utils/backup/download/:name/:created_at 	GET 	Download a backup
+  def download
+    @backup = Backup.new(
+      name: params[:name],
+      created_at: params[:created_at]
+    )
+    respond_to do |format|
+      if @backup.exist?
+        format.any do
+          send_file(
+            @backup.path,
+            filename: "#{params[:name]}-#{params[:created_at]}.tar.gz"
+          )
+        end
+      else
+        msg = I18n.t(".missing_backup", scope: "backup.index")
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+        format.json { render json: msg }
+      end
+    end
+  end
+
+  #
+  # Upload
+  #
+  # Provides the restful api call for
+  # /utils/backup/upload   POST   Upload a backup
+  def upload
+    file = params[:fileupload]
+    filename, created_at = Backup.filename_time(file.original_filename)
```

Comment:
Make `initialize` more flexible to accept name and created_at OR filename. Than you can move that into the model.